### ISSUE TITLE
StartFlowRun return flow_id after waiting on run.

### ISSUE THREAD

**Issue by peterroelants** [PrefectHQ/prefect#4625]
## Current behavior

Currently `StartFlowRun` only returns the `flow_run_id` directly when not waiting [[reference](https://github.com/PrefectHQ/prefect/blob/1867f205405e1f68b828ce9dcf00bbd4969903bc/src/prefect/tasks/prefect/flow_run.py#L195)].

When waiting for the run to complete by using `StartFlowRun(..., wait=True)` the result doesn't hold the `flow_run_id` as a separate object, and the id needs to be parsed from the signal exception message, for example by a separate task:
```
def parse_flow_id(flow_run_result):
    flow_id = str(flow_run_result).split(" ")[0]
```
Example based on [this slack message](https://prefect-community.slack.com/archives/CL09KU1K7/p1623086665045800?thread_ts=1622645858.327700&cid=CL09KU1K7). Signal exception message is being set at [`flow_run.py#L202`](https://github.com/PrefectHQ/prefect/blob/1867f205405e1f68b828ce9dcf00bbd4969903bc/src/prefect/tasks/prefect/flow_run.py#L202).

Extracting the `flow_run_id` feels a bit hacky, and is prone to fail when the message string changes slightly.

## Proposed behavior

Add the `flow_run_id` directly the signal.

This would either require:
* The ability to set various objects on Prefect signal states.
* Dedicated state for flow runs that contains the `flow_run_id`
* Change the logic of running and waiting for a flow run, to only raise an exception when the flow was failed, and return the `flow_run_id` on success, similar to what happens when not waiting.
* ...

## Example

The `flow_run_id` is needed to fetch the result of a flow run as Task as illustrated by [this Slack post](https://prefect-community.slack.com/archives/CL09KU1K7/p1623086665045800?thread_ts=1622645858.327700&cid=CL09KU1K7). Fetching a result from a flow run is currently a bit of a hassle, this small change would make it less hacky and slightly more intuitive.

**Comment by madkinsz** [comment]
Hey @peterroelants -- thanks for the report.

This is something I addressed in https://github.com/PrefectHQ/prefect/pull/4245 but we decided not to change that interface since it could be considered "breaking". You may want to just take that code. A redesign of flow run tasks is coming in #4434 / #4563 which will very likely be in the next release. In this task, we'll always return a flow run id for simplicity.

I'm going to close this since it's a known issue that is already being addressed. Let me know if you have any more questions though!
